doc/auth.md

```diff
@@ -0,0 +1,320 @@
+# Kuadrant Auth
+
+A Kuadrant AuthPolicy custom resource:
+
+1. Allows it to target Gateway API networking resources such as [HTTPRoutes](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRoute) and [Gateways](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.Gateway), using these resources to obtain additional context, i.e., which traffic workload (HTTP attributes, hostnames, user attributes, etc) to enforce auth.
+2. Allows to specify which specific subsets of the targeted network resource to apply the auth rules to.
```

Comment:
The start of these points reads funny to me. I wonder if the suggested is better. 
```suggestion
1. Targets Gateway API networking resources such as [HTTPRoutes](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRoute) and [Gateways](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.Gateway), using these resources to obtain additional context, i.e., which traffic workload (HTTP attributes, hostnames, user attributes, etc) to enforce auth.
2. Specify which specific subsets of the targeted network resource to apply the auth rules to.
```

Reply:
I can cope with the suggestion for bullet point 1 (to be ported to https://github.com/Kuadrant/kuadrant-operator/blob/main/doc/rate-limiting.md as well), but 2 needs to be clear that specific subsets is optional.

Reply:
If point 2 needs to be clear that specific subsets are optional I think both version are not clear. Possible might need to use the word optional in there.